conans/client/generators/__init__.py

```diff
@@ -138,6 +175,15 @@ def write_toolchain(conanfile, path, output):
                 raise ConanException("Unknown toolchain '%s'" % conanfile.toolchain)
             tc = toolchain(conanfile)
             with chdir(path):
-                tc.write_toolchain_files()
+                tc.generate()
+
+        # TODO: Lets discuss what to do with the environment
+
+    if hasattr(conanfile, "generate"):
+        assert callable(conanfile.generate), "generate should be a method, not an attribute"
```

Comment:
We are not doing this for other methods. The Python error itself and the `conanfile_exception_formatter` should be enough.

```suggestion
```